presto-main/src/test/java/com/facebook/presto/operator/scalar/BenchmarkArrayIntersect.java

```diff
@@ -0,0 +1,195 @@
+        private String type = "BIGINT";
+
+        @Param({"10", "100", "1000"})
+        private int arraySize = 10;
+
+        private Page page;
+        private PageProcessor pageProcessor;
+
+        @Setup
+        public void setup()
+        {
+            Type elementType;
+            switch (type) {
+                case "BIGINT":
+                    elementType = BIGINT;
+                    break;
+                case "VARCHAR":
+                    elementType = VARCHAR;
+                    break;
+                case "DOUBLE":
+                    elementType = DOUBLE;
+                    break;
+                case "BOOLEAN":
+                    elementType = BOOLEAN;
+                    break;
+                default:
+                    throw new UnsupportedOperationException();
+            }
+
+            ArrayType arrayType = new ArrayType(elementType);
+            Signature signature = new Signature(name, FunctionKind.SCALAR, arrayType.getTypeSignature(), arrayType.getTypeSignature(), arrayType.getTypeSignature());
+            ImmutableList<RowExpression> projections = ImmutableList.of(
+                    new CallExpression(signature, arrayType, ImmutableList.of(field(0, arrayType), field(1, arrayType))));
+
+            MetadataManager metadata = MetadataManager.createTestMetadataManager();
+            ExpressionCompiler compiler = new ExpressionCompiler(metadata, new PageFunctionCompiler(metadata, 0));
+            pageProcessor = compiler.compilePageProcessor(Optional.empty(), projections).get();
+
+            page = new Page(createChannel(POSITIONS, arraySize, elementType), createChannel(POSITIONS, arraySize, elementType));
+        }
+
+        private static Block createChannel(int positionCount, int arraySize, Type elementType)
+        {
+            ArrayType arrayType = new ArrayType(elementType);
+            BlockBuilder blockBuilder = arrayType.createBlockBuilder(null, positionCount);
+            for (int position = 0; position < positionCount; position++) {
+                BlockBuilder entryBuilder = blockBuilder.beginBlockEntry();
+                for (int i = 0; i < arraySize; i++) {
+                    if (elementType.getJavaType() == long.class) {
+                        elementType.writeLong(entryBuilder, ThreadLocalRandom.current().nextLong() % arraySize);
+                    }
+                    else if (elementType.getJavaType() == double.class) {
+                        elementType.writeDouble(entryBuilder, ThreadLocalRandom.current().nextDouble() % arraySize);
+                    }
+                    else if (elementType.getJavaType() == boolean.class) {
+                        elementType.writeBoolean(entryBuilder, ThreadLocalRandom.current().nextBoolean());
+                    }
+                    else if (elementType.equals(VARCHAR)) {
+                        // make sure the size of a varchar is rather small; otherwise the aggregated slice may overflow
```

Comment:
Each string is at most |arraySize| length. If arraySize = 100, then the strings created would be "0" to "99". It is made this way to avoid the 4MB limit in TypedSet elementBlock to be reached. If the string is very long and all distinct, the 4MB limit could be reached.